VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleListEntryController.cs

```diff
@@ -116,6 +120,66 @@ public IHttpActionResult CreateLinks(webModel.ListEntryLink[] links)
             return StatusCode(HttpStatusCode.NoContent);
         }
 
+        /// <summary>
+        /// Bulk create links to categories and items
+        /// </summary>
+        /// <param name="createLinksModel"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("~/api/catalog/listentrylinks/bulkcreate")]
+        [ResponseType(typeof(void))]
+        public IHttpActionResult BulkCreationLinks(webModel.BulkCreateLinkParameters createLinksModel)
+        {
+            var coreModelCriteria = createLinksModel.SearchCriteria.ToCoreModel();
+
+            bool haveProductsOrCategories;
+
+            do
+            {
+                var links = new List<webModel.ListEntryLink>();
+
+                var searchResult = _searchService.Search(coreModelCriteria);
+
+                var productLinks = searchResult
```

Comment:
There is the code duplication that can be avoided.
Try to define three new constructors  in ListEntryLink type  
```
ListEntryLink(Product product, string catalogId, string categoryId)
ListEntryLink(Category category, string catalogId, string categoryId)
ListEntryLink(string type , string catalogId, string categoryId)
```